File: tools/combine.rb
#!/usr/bin/env ruby

CUTOFF = 275 #Float::INFINITY

$dict = {}
File.open('/usr/local/google/home/kjs/wordlists/data/csw.txt').each_line do |line|
  line.chomp!
  word, defn = line.split(' ', 2)
  $dict[word] = defn
end

def output(same, num)
  defns = same.map {|w| "#{w} = #{$dict[w]}"}.join("\n")
  puts "#{num}: #{same.join(' ')}\n-----------------\n#{defns}\n\n"
end

words = 0
same = []
num = 0
ARGF.each_line do |line|
  break if num >= CUTOFF
  line.chomp!
  if !same.empty?
    if same[0].chars.sort.join != line.chars.sort.join
      num += 1
      output(same, num)
      words += same.length
      same = [line]
      next
    end
  end
  same << line
end
num += 1
output(same, num) if num < CUTOFF
#p words

File: tools/combine.js
#!/usr/bin/env node
'use strict';

const fs = require('fs');
const path = require('path');

const stats = {};
const results = path.join(__dirname, 'results');
for (const f of fs.readdirSync(results)) {
  if (!f.endsWith('.json')) continue;

  const dice = f.split('-')[0];
  stats[dice] = stats[dice] || {};

  const data = JSON.parse(fs.readFileSync(path.join(results, f)));
  for (const word in data) {
    stats[dice][word] = (stats[dice][word] || 0) + (data[word] * score(word));
  }
}

for (const dice in stats) {
  const words = Object.entries(stats[dice]).sort((a, b) => b[1] - a[1]).map(e => e.join(' ')).join('\n');
  fs.writeFileSync(path.join(results, `${dice.toLowerCase()}.txt`), words);
}

function score(word) {
  if (word.length < 3) return 0;
  if (word.length <= 4) return 1;
  if (word.length === 5) return 2;
  if (word.length === 6) return 3;
  if (word.length === 7) return 5;
  /* if (word.length >= 8) */ return 11;
}
